Earthfile

```diff
@@ -873,3 +874,32 @@ check-broken-links:
     ELSE
         RUN --no-cache echo -e "${GREEN}No Broken Links were found${NOCOLOR}"
     END
+
+# open-pr-for-fork creates a new PR based on the given pr_number
+open-pr-for-fork:
+    ARG git_repo="earthly/earthly"
+    ARG git_url="git@github.com:$git_repo"
+    ARG earthly_lib_version=2.2.2
+    DO github.com/earthly/lib/ssh:$earthly_lib_version+ADD_KNOWN_HOSTS --target_file=~/.ssh/known_hosts
+    RUN git config --global user.name "littleredcorvette" && \
+        git config --global user.email "littleredcorvette@users.noreply.github.com"
+    GIT CLONE "$git_url" earthly
+    WORKDIR earthly
+    ARG git_hash=$(git rev-parse HEAD)
+    RUN --mount=type=secret,id=littleredcorvette-id_rsa,mode=0400,target=/root/.ssh/id_rsa \
+        git fetch --unshallow
+    ARG TARGETARCH
+    # renovate: datasource=github-releases depName=cli/cli
+    ARG gh_version=v2.36.0
+    RUN curl -Lo ghlinux.tar.gz \
+      https://github.com/cli/cli/releases/download/$gh_version/gh_${gh_version#v}_linux_${TARGETARCH}.tar.gz \
+      && tar --strip-components=1 -xf ghlinux.tar.gz \
+      && rm ghlinux.tar.gz
+    ARG --required pr_number
+    RUN --no-cache --mount=type=secret,id=littleredcorvette-id_rsa,mode=0400,target=/root/.ssh/id_rsa \
+        --secret GH_TOKEN=littleredcorvette-github-token \
+        ./bin/gh pr checkout $pr_number --branch "test-pr-$pr_number" && \
+        git merge origin/main && \
```

Comment:
you might also need to have a 

```
git commit --allow-empty -m "please run the test"
```

for cases where no merge is required.

this is needed because github actions won't re-run the job if a previous job has already been started on the same commit.